From b36800a56775a0f9a8cada6225513aef2e5c1fdf Mon Sep 17 00:00:00 2001
From: Aradhya Bhatia <a-bhatia1@ti.com>
Date: Fri, 9 Dec 2022 09:21:15 -0600
Subject: [PATCH 1/4] TMP: drm/tidss: introduce support for tidss reversion
 check

Add a function to lookup the subrevision of the tidss component. this
can be used to introduce a hack later in the series.

Signed-off-by: Aradhya Bhatia <a-bhatia1@ti.com>
Signed-off-by: Nishanth Menon <nm@ti.com>
---
 drivers/gpu/drm/tidss/tidss_dispc.c | 7 +++++++
 drivers/gpu/drm/tidss/tidss_dispc.h | 1 +
 2 files changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/tidss/tidss_dispc.c b/drivers/gpu/drm/tidss/tidss_dispc.c
index cea9f443e01f..cfc41e20dc79 100644
--- a/drivers/gpu/drm/tidss/tidss_dispc.c
+++ b/drivers/gpu/drm/tidss/tidss_dispc.c
@@ -2035,6 +2035,13 @@ const u32 *dispc_plane_formats(struct dispc_device *dispc, unsigned int *len)
 	return dispc->fourccs;
 }
 
+enum dispc_dss_subrevision dispc_get_subrev(struct dispc_device *dispc)
+{
+	WARN_ON(!dispc->feat->subrev);
+
+	return dispc->feat->subrev;
+}
+
 static s32 pixinc(int pixels, u8 ps)
 {
 	if (pixels == 1)
diff --git a/drivers/gpu/drm/tidss/tidss_dispc.h b/drivers/gpu/drm/tidss/tidss_dispc.h
index 6cb0a6ab6eb4..b7b95f81474c 100644
--- a/drivers/gpu/drm/tidss/tidss_dispc.h
+++ b/drivers/gpu/drm/tidss/tidss_dispc.h
@@ -137,6 +137,7 @@ int dispc_plane_setup(struct dispc_device *dispc, u32 hw_plane,
 		      u32 hw_videoport);
 int dispc_plane_enable(struct dispc_device *dispc, u32 hw_plane, bool enable);
 const u32 *dispc_plane_formats(struct dispc_device *dispc, unsigned int *len);
+enum dispc_dss_subrevision dispc_get_subrev(struct dispc_device *dispc);
 
 int dispc_init(struct tidss_device *tidss);
 void dispc_remove(struct tidss_device *tidss);
-- 
2.30.2

